### ISSUE TITLE
[BUG] - [94. Binary Tree Inorder Traversal] Unclear examples and theoretical basis

### ISSUE THREAD

**Issue by warcayac** [LeetCode-Feedback/LeetCode-Feedback#23611]
### LeetCode Username

warcayac

### Problem Number, Title, and Link

https://leetcode.com/problems/binary-tree-inorder-traversal/description/

### Bug Category

Problem description, Problem examples

### Bug Description

There is a serious problem about how you describe the problem and how you expect the problem to be solved by using a problem some more complex. This is because there seems to be no consensus on how to apply the Traversal algorithms.

For the first example in the description of problem you set as entry this array `[1,null,2,3]` but you don't mention what notation (pre-order or post-order) represents this array, and if it weren't for the attached image one could work with the wrong notation. Well, now we know what notation you are using: **pre-order traversal**. Next question would be: **how are you applying it?** This is where the lack of a more complex example is necessary.

Because this topic is somewhat new to me I was researching over books and internet and I found different approaches on how to apply this algorithm.

1. Data Structures & Algorithms in Dart by Jonathan Sande, 1st ed, 2022, page 117:
![image](https://github.com/user-attachments/assets/ee8cfe93-871e-4bd4-ae80-b27ec32ea2ac)
If I apply these values in your test I get:
![image](https://github.com/user-attachments/assets/323e994f-21b3-4f4c-8f89-a0aad19361e2)
They are not the same.
2. JavaScript Data Structures and Algorithms by Sammie Bae, 2019, pages 207-209:
![image](https://github.com/user-attachments/assets/f4d8b04a-cd9b-452f-a2be-1496243d8a51)
![image](https://github.com/user-attachments/assets/5e97c6ba-352a-4a41-b6fb-9675ce953947)
If I apply these values in your test I get:
![image](https://github.com/user-attachments/assets/dd7c1f89-4959-4bb6-85a0-5dc80a683439)
They are not the same.

3. https://www.geeksforgeeks.org/tree-traversals-inorder-preorder-and-postorder/
![image](https://github.com/user-attachments/assets/315d9c6e-2186-4935-a1bc-87b5a1bad527)
If I apply these values in your test I get:
![image](https://github.com/user-attachments/assets/7b9448ce-164c-4f06-b91f-fb9857712079)
They are not the same.

So the natural question is: **Who's wrong?** or **who do I trust?**

Another concern in your problem is about format of your output (solution) because it is not coherent with the input format which uses **null** values to indicate the end of a branch. Why do you use NULL for input but not for output? Every output should be used to re-create the original binary tree, but with your output format it brings inaccuracies. For example, how do you create a binary tree from this array **[4, 2, 8, 5, 9, 1, 6, 3, 7]** which uses in-order traversal? Without an algorithm defined and clear and the use of NULLs, it could have multiple representations.

In short, I think your problem is set poorly out.

**NOTE**: it seems you are using **Level-Order Traversal** (_Breadth-First Search_) for your output which is not **inorder traversal** as your problem requests.

### Language Used for Code

None

### Expected behavior

An array which can be used to re-create precisely the original binary tree.

**Comment by warcayac** [comment]
I understand NOW:

- what input format you're using: level-order form, and 
- output format represents the requested traversal only,

However, why were these descriptions not included along the problem statement? in other previous problems you are more descriptive in your statements even in examples, but it doesn't occur here, it's lacking.

Another concern you aren't addressing is about the need of an example more complex bringing better understanding on how you expect the problem to be solved, and this is a request made by other users in your forum and here also, However, it seems that you do not want to attend to them.
